File: export-templates/python/agent.py
import argparse
import pyfiglet
from components.http_server import start_http_server
from components.cli_chat import start_cli_chat

if __name__ == "__main__":    
    print(pyfiglet.figlet_format("FAQtiv"), flush=True)

    parser = argparse.ArgumentParser(description="FAQtiv Agent CLI/HTTP Server")
    parser.add_argument("--http", action="store_true", help="Run as HTTP server")
    args = parser.parse_args()

    if args.http:
        start_http_server()
    else:
        start_cli_chat()

File: export-templates/javascript/agent.js
const figlet = require('figlet');
const { startHttpServer } = require('./components/http-server');
const { startCliChat } = require('./components/cli-chat');
const { ENV_VARS } = require('./constants');

if (require.main === module) {
  console.log(figlet.textSync("FAQtiv"));

  for (const [key, value] of Object.entries(ENV_VARS)) {
    process.env[key] = value;
  }

  const args = process.argv.slice(2);
  if (args.includes('--http')) {
    startHttpServer();
  } else {
    startCliChat();
  }
}
